src/OpenLoco/src/GameCommands/Terraform/LowerRaiseLandMountain.cpp

```diff
@@ -0,0 +1,405 @@
+            *mtnToolHeightDiff += kSmallZStep;
+            if (targetPos.y < args.pointA.y)
+            {
+                *mtnToolHeightDiff += kSmallZStep;
+            }
+        }
+
+        height = TileManager::getSurfaceCornerHeight(*surface, SurfaceSlope::CornerUp::east);
+        adjustSurfaceSlopeSouth(targetPos, height, removedBuildings);
+    }
+
+    static void smoothenSurfaceWest(const Pos2& refPos, const Pos2& targetPos, const LowerRaiseLandMountainArgs& args, std::set<Pos3, LessThanPos3>& removedBuildings)
+    {
+        auto tile = TileManager::get(refPos);
+        auto surface = tile.surface();
+
+        auto height = TileManager::getSurfaceCornerHeight(*surface, SurfaceSlope::CornerUp::east);
+        adjustSurfaceSlopeNorth(targetPos, height, removedBuildings);
+
+        *mtnToolHeightDiff -= kSmallZStep;
+
+        if (targetPos.x <= args.pointB.x)
+        {
+            *mtnToolHeightDiff += kSmallZStep;
+            if (targetPos.x < args.pointA.x)
+            {
+                *mtnToolHeightDiff += kSmallZStep;
+            }
+        }
+
+        height = TileManager::getSurfaceCornerHeight(*surface, SurfaceSlope::CornerUp::south);
+        adjustSurfaceSlopeWest(targetPos, height, removedBuildings);
+    }
+
+    // 0x00462DCE
+    static uint32_t lowerRaiseLandMountain(const LowerRaiseLandMountainArgs& args, const uint8_t flags)
+    {
+        mtnToolXParams = (args.pointB.x << 16) | args.pointA.x;
+        mtnToolYParams = (args.pointB.y << 16) | args.pointA.y;
+        mtnToolGCFlags = flags;
+
+        if (flags & Flags::apply)
+        {
+            S5::getOptions().madeAnyChanges = 1;
+        }
+
+        mtnToolCost = 0;
+
+        // We keep track of removed buildings for each tile visited
+        // this prevents accidentally double counting their removal
+        // cost if they span across multiple tiles.
+        std::set<World::Pos3, LessThanPos3> removedBuildings{};
+
+        // Play sound if this is a company action
+        if ((flags & Flags::apply) && getCommandNestLevel() == 1 && getUpdatingCompanyId() != CompanyId::neutral)
+        {
+            const auto height = TileManager::getHeight(args.centre).landHeight;
+            Audio::playSound(Audio::SoundId::construct, World::Pos3(args.centre.x, args.centre.y, height));
+        }
+
+        // First, raise/lower the mountain's centre tile
+        {
+            auto result = adjustMountainCentre(args, removedBuildings, flags);
+            if (result != FAILURE)
+            {
+                mtnToolCost = *mtnToolCost + result;
+            }
+        }
+
+        if (!validCoords(args.pointA) || !validCoords(args.pointB))
+        {
+            return FAILURE;
+        }
+
+        // 0x00462E7E
+        auto preTile = TileManager::get(args.pointA);
+        auto* preSurface = preTile.surface();
+        if (preSurface->slope() != 0)
+        {
+            auto result = adjustMountainCentre(args, removedBuildings, flags);
```

Comment:
vanilla looks to flip this. If there was a slope it does the opposite of the direction of change.

Reply:
Hmm, okay. I'm not sure what requires adjustment though. Could you elaborate?

Reply:
Its more of an observation could you just run a few tests with the mountain tool centre being none-flat and see if we match vanilla.

Reply:
I did a bit of testing, and haven't been able to observe any differences. Maybe worth investigating and revisiting in the future, but nothing to stop me from merging now, I think.